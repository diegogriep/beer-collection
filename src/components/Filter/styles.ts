import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    color: #fff;
    padding-top: ${theme.spacings.xxsmall};

    > a {
      color: #fff;
      display: block;
      font-size: ${theme.font.sizes.xsmall};
      position: absolute;
      right: ${theme.spacings.xxsmall};
      top: ${theme.spacings.xxsmall};
    }

    form {
      display: flex;

      button {
        border-radius: 0 ${theme.border.radius} ${theme.border.radius} 0;
      }
    }

    label {
      display: none;
      font-size: ${theme.font.sizes.xsmall};
    }

    input {
      border: 0;
      border-radius: ${theme.border.radius} 0 0 ${theme.border.radius};
      color: ${theme.colors.mainBg};
      cursor: pointer;
      flex: 1;
      font: ${theme.font.bold} ${theme.font.sizes.xsmall} ${theme.font.family};
      padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
      text-decoration: none;
    }

    ${media.greaterThan('small')`
      align-items: center;
      column-gap: ${theme.spacings.small};
      display: flex;
      justify-content: center;
      padding-top: ${theme.spacings.medium};

      label {
        line-height: 3.4rem;
        padding-right: ${theme.spacings.xsmall};
      }

      input {
        flex: none;
        width: 25rem;
      }
    `}
  `}
`
