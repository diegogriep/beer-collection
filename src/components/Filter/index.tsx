'use client'

import { FormEvent, useState } from 'react'
import * as S from './styles'
import { useRouter } from 'next/navigation'
import Button from '../Button'

const Filter = () => {
  const [value, setValue] = useState('')
  const onChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    setValue(e.currentTarget.value)
  }

  const { push } = useRouter()

  async function onSubmit(event: FormEvent<HTMLFormElement>) {
    event.preventDefault()
    push(`/?beer_name=${value}`)
  }
  return (
    <S.Wrapper>
      <h4>Filter by name</h4>
      <form onSubmit={onSubmit}>
        <input onChange={onChange} type="text" name="beer_name" placeholder='What beer are you looking for?' />
        <Button type="submit">
          Submit
        </Button>
      </form>
    </S.Wrapper>
  )
}

export default Filter
