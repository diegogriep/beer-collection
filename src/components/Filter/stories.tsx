import { Meta, StoryObj } from '@storybook/react'
import Filter from '.'

export default {
  title: 'components/Filter',
  component: Filter
} as Meta

export const Default: StoryObj = {}
