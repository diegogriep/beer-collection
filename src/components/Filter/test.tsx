import { screen } from '@testing-library/react'

import Filter from '.'
import { renderWithTheme } from '../../utils/tests/helpers'

jest.mock('next/navigation', () => ({
  usePathname: jest.fn()
}))

describe('<Filter />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Filter />)

    expect(screen.getByText(/Filter by name/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
