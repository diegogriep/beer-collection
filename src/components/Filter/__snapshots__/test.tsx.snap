// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render the heading 1`] = `
.c0 {
  color: #fff;
  padding-top: 0.8rem;
}

.c0 >a {
  color: #fff;
  display: block;
  font-size: 1.2rem;
  position: absolute;
  right: 0.8rem;
  top: 0.8rem;
}

.c0 form {
  display: flex;
}

.c0 form button {
  border-radius: 0 0.4rem 0.4rem 0;
}

.c0 label {
  display: none;
  font-size: 1.2rem;
}

.c0 input {
  border: 0;
  border-radius: 0.4rem 0 0 0.4rem;
  color: #C96E12;
  cursor: pointer;
  flex: 1;
  font: 700 1.2rem mocked,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,sans-serif;
  padding: 0.8rem 1.6rem;
  text-decoration: none;
}

.c1 {
  background: linear-gradient(180deg, #C96E12 0%, #DF8D03 50%);
  border: 0;
  border-radius: 0.4rem;
  color: #fff;
  display: inline-flex;
  cursor: pointer;
  font: 700 1.6rem mocked,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,sans-serif;
  padding: 0.8rem 1.6rem;
  text-decoration: none;
}

.c1:hover {
  background: linear-gradient(180deg, #DF8D03 0%, #C96E12 50%);
}

@media (min-width: 450px) {
  .c0 {
    align-items: center;
    column-gap: 2.4rem;
    display: flex;
    justify-content: center;
    padding-top: 3.2rem;
  }

  .c0 label {
    line-height: 3.4rem;
    padding-right: 1.6rem;
  }

  .c0 input {
    flex: none;
    width: 25rem;
  }
}

<div
  class="c0"
>
  <a
    href="/add"
    title="add new bottle"
  >
    add new bottle
  </a>
  <form>
    <label
      for="beer_name"
    >
      Filter by name
    </label>
    <input
      id="beer_name"
      name="beer_name"
      placeholder="What beer are you looking for?"
      type="text"
    />
    <button
      class="c1"
      type="submit"
    >
      Search
    </button>
  </form>
</div>
`;
